src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1038,7 +1038,7 @@ MethodDefinition GetMethodWithNoParameters (TypeDefinition type, string methodna
 				if (method != null)
 					return method;
 
-				type = type.BaseType.Resolve ();
+				type = type.BaseType?.Resolve ();
```

Comment:
I don't necessarily know if this fixes the issue, but this looks like an outlier compared to:

* https://github.com/mono/linker/blob/master/src/linker/Linker.Steps/MarkStep.cs#L1021-L1028
* https://github.com/mono/linker/blob/master/src/linker/Linker.Steps/MarkStep.cs#L985-L992